site-src/geps/gep-1058.md

```diff
@@ -0,0 +1,962 @@
+        value: /store
+    filters:
+      - type: RequestHeaderModifier
+        requestHeaderModifier:
+          add:
+            - name: my-source
+              value: store
+    allowedRoutes:
+      namespaces:
+        from: Selector
+        selector:
+          matchExpressions:
+          - key: kubernetes.io/metadata.name
+            operator: In
+            values: [store-ns, core-svces-ns]
+  - name: api
+    matches:
+    - path:
+        type: PathPrefix
+        value: /api
+    allowedRoutes:
+      namespaces:
+        from: Selector
+        selector:
+        matchExpressions:
+        - key: kubernetes.io/metadata.name
+          operator: In
+          values: [api-ns, core-svces-ns]
+status:
+  parents:
+  - parentRef:
+      name: prod-web-gw
+    controllerName: prod-gw
+    conditions:
+    - type: Accepted
+      status: true
+      reason: Accepted
+      Message: Route successfully attached to Gateway.
+---
+kind: HTTPRoute
+metadata:
+  name: store-route
+  namespace: store-ns
+spec:
+  parentRefs:
+    - kind: HTTPRoute
+      name: my-example
+      namespace: prod-gw
+      sectionName: store
+  rules:
+  - matches:
+    - path:
+        type: PathPrefix
+        value: /images
+    backendRefs:
+    - name: image-service
+      port: 8080
+  - matches:
+    - path:
+        type: PathPrefix
+        value: /cart
+    backendRefs:
+    - name: cart-service
+      port: 8080
+status:
+  parents:
+    - parentRef:
+        kind: HTTPRoute
+        name: my-example
+        namespace: prod-gw
+        sectionName: store
+      controllerName: prod-gw
+      conditions:
+      - type: Attached
+        status: true
+        reason: Attached
+        message: Successfully attached to `store` section.
+      grandparents:
+      - parentRef:
+          kind: Gateway
+          name: prod-web-gw
+        conditions:
+        - type: Accepted
+          status: true
+          reason: Accepted
+          Message: Route successfully attached to parent and gateway.
+---
+kind: HTTPRoute
+metadata:
+  name: core-svces
+  namespace: core-svces-ns
+spec:
+  parentRefs:
+  - kind: HTTPRoute
+    name: my-example
+    namespace: prod-gw
+    sectionName: store
+  - kind: HTTPRoute
+    name: my-example
+    namespace: prod-gw
+    sectionName: api
+  rules:
+  - matches:
+    - path:
+        type: PathPrefix
+        value: /login
+    backendRefs:
+    - name: auth-service
+      port: 8080
+  - matches:
+    - path:
+        type: Pathprefix
+        value: /account
+    backendRefs:
+    - name: account-service
+      port: 8080
+status:
+  parents:
+  - parentRef:
+      kind: HTTPRoute
+      name: my-example
+      namespace: prod-gw
+      sectionName: store
+    controllerName: prod-gw
+    conditions:
+    - type: Attached
+      status: true
+      reason: Attached
+      message: Successfully attached to `store` section.      
+    grandparents:
+    - parentRef:
+          kind: Gateway
```

Comment:
I am not convinced this is something the child route needs to know about. 

For example, when I bind a HTTPRoute to a Gateway, that Gateway binds to a GatewayClass and that gateway class to a controller, and that controller probably sets up Foo and Bar resources in the process. But HTTPRoute doesn't care about any of that -- it just needs to know if it attached to the Gateway or not.

Reply:
I think as a Route owner it would be very useful to know where my Route is exposed without having to traverse a tree. With that said, I recognize that this could become complex to implement and can be added separately from the rest of the delegation API. Maybe this particular change (grandparents in status) could be pulled off into another dependent GEP so it could be reviewed independently if necessary?

Reply:
Yes, I think the concept needs work, and it's probably worth implementing the rest and seeing what would be useful to have in each status.

The overall design goal here is for the owner of the child Route not to need to also fetch the parent Route and the Gateway to determine if their thing is being exposed correctly.

Reply:
That definitely seems like a valid use case. It actually brings up a bit of a gap in Gateway<->Route. Can a route succesfully bind to a Gateway, while the Gateway itself is not Ready? It seems like possibly that should be indicated (strawman: `Attached=false, Reason=parentNotReady` or something). A similar concept could be used here.

The key difference I am trying to make here is its a ~boolean - we just say the parent is ready or not, not all the details. This ties in to the fanout concerns mentioned below as well

Reply:
I think adding an `Attached` RouteConditionType is actually a _great_ fit for this - the similar `ListenerConditionType` already describes almost this exact scenario, including differentiation from `Accepted` (rejected from _immediate_ parentRef target due to invalid syntax/semantics, such as trying to delegate to a TCPRoute):
> This condition indicates that, even though the listener is syntactically and semantically valid, the controller is not able to configure it on the underlying Gateway infrastructure.